scripts/migrate_permissions.ts

```diff
@@ -0,0 +1,70 @@
+import 'dotenv/config'
+import {
+    newClient,
+    readArtifact,
+    executeContract,
+    updateContractAdmin,
+} from './helpers.js'
+import {LCDClient, Wallet} from "@terra-money/terra.js";
+
+async function proposeNewOwner(terra: LCDClient, wallet: Wallet, newOwner: string, contractAddress: string) {
+    try {
+        await executeContract(terra, wallet, contractAddress, {
+            propose_new_owner: {
+                owner: newOwner,
+                expires_in: 1234567
+            }
+        });
+    } catch (e: any) {
+        console.log(e.response.data.message)
+    }
+}
+
+async function claimOwnership(terra: LCDClient, wallet: Wallet, contractAddress: string) {
+    try {
+        await executeContract(terra, wallet, contractAddress, {
+            claim_ownership: {}
+        });
+    } catch (e: any) {
+        console.log(e.response.data.message)
+    }
+}
+
+async function updateAdmin(terra: LCDClient, wallet: Wallet, newAdminAddress: string, contractAddress: string) {
+    try {
+        await updateContractAdmin(terra, wallet, newAdminAddress, contractAddress);
+    } catch (e: any) {
+        console.log(e.response.data.message)
+    }
+}
+
+async function main() {
+    const { terra, wallet } = newClient()
+    console.log(`chainID: ${terra.config.chainID} wallet: ${wallet.key.accAddress}`)
+    const network = readArtifact(terra.config.chainID)
+    console.log('network:', network)
+
+    // create propose new owner for our contracts
+    for (const key in network) {
```

Comment:
We need to define each contracts manually as network.json contains not only contract addresses

Reply:
Why? For example, if it is not the address of the contract, we will catch an error, skip this address and continue work. Also, we can specify a file that includes only the necessary contract addresses.

Reply:
I don't see where we are catching errors in the current code

Reply:
Ok, catching error is implicit but it should work. You are correct